src/main/java/uk/gov/companieshouse/api/accounts/exceptionhandler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.companieshouse.api.accounts.exceptionhandler;
+
+import java.security.NoSuchAlgorithmException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value = NoSuchAlgorithmException.class)
```

Comment:
I think it would be good to add the Exception.class too. Devs seeing this class existing with global in the name might assume it's catching all exceptions.